app/src/layout.py

```diff
@@ -0,0 +1,33 @@
+import json
+
+from panel.layout.flex import FlexBox
+
+from .config_importer import ConfigImporter
+from .plotter import PlotterFactory
+
+
+# Load data
+with open("data/outfile.json", "r") as f:
+    data = json.load(f)
+
+# Load config
+config_importer = ConfigImporter()
+config = config_importer.get_config()
+
+# Initialize plotter factory
+plotter_factory = PlotterFactory()
+
+chart_collection = {}
+for plot_key in config:
+    # Generate plots and filters
+    plotter = plotter_factory.create_plotter(data[plot_key], config[plot_key])
+    plotter.generate()
+
+    # Place plots and filters in the layout
+    flex_obj = FlexBox(*list(plotter.plots.values()),
+                       *list(plotter.filters.values()),
+                       flex_direction="column",
+                       align_items="center",
+                       sizing_mode="stretch_width")
+
+    chart_collection[plot_key] = flex_obj
```

Comment:
More generally, as this is an operation that has to be done only once, at startup of the process, I would do this:
1) Check, but I'm pretty sure, all of this (including IO) has to be done for every page request. If confirmed, save some timings of first and subsequent loading times and then:
2) Put all of this code in a function returning the chart_collection, then decorate it with `@lru_cache` https://docs.python.org/3/library/functools.html#functools.lru_cache use that function from main.py and then redo the timings. The first call should be minimal longer, but the next ones quite faster, even more on the server

Reply:
I checked and it seems that it's actually only done once at the startup